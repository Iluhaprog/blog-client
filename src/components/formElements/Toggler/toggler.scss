@import '../../../vars.scss';

.toggler {
    &-box {
        width: 50px;
        padding: 2px;
        background-color: #fff;
        border: $border-1;
        border-width: 2px;
        border-radius: 15px;
        
        &__item {
            width: 20px;
            height: 20px;
            background-color: $text-color-3;
            border-radius: 50%;
            margin-left: 0;
            transition: .2s ease-in-out;

            &_right {
                margin-left: calc(100% - 20px);
            }
        }
    }

}