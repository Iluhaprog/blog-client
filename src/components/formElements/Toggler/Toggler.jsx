import React, { useState, useEffect } from 'react';
import './toggler.scss';

export default props => {
    const [value, setValue] = useState(false);

    useEffect(() => {
        setValue(props.value);
    }, [props]);

    const handleClick = () => {
        props.onChange(!value);
    }

    return (
        <div className='toggler' onClick={handleClick}>
            <div className='toggler-box'>
                <div 
                    className={
                        `toggler-box__item ${value ? 'toggler-box__item_right' : ''}`
                    }
                >
                </div>
            </div>
        </div>
    );
}