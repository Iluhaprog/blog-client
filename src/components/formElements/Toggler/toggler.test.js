import React from 'react'
import { render, fireEvent, waitFor } from '@testing-library/react';
import Toggler from './Toggler';

describe('Test view of toggler', () => {
    test('Should show a toggler with "false" value', () => {
        const { asFragment } = render(<Toggler value={false} onChange={() => {}}/>);
        expect(asFragment()).toMatchSnapshot();
    });

    test('Should show a toggler with "true" value', () => {
        const { asFragment } = render(<Toggler value={true} onChange={() => {}}/>);
        expect(asFragment()).toMatchSnapshot();
    });
});

describe('Test events of toggler', () => {
    test('Should handle onchange event', async () => {
        const handleChange = jest.fn();
        const { container } = render(<Toggler value={false} onChange={handleChange}/>);
        fireEvent.click(container.querySelector('.toggler'));
        await waitFor(() => {
            expect(handleChange).toHaveBeenCalledTimes(1);
        });
    });
});