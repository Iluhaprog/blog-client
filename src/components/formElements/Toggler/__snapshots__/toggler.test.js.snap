// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test view of toggler Should show a toggler with "false" value 1`] = `
<DocumentFragment>
  <div
    class="toggler"
  >
    <div
      class="toggler-box"
    >
      <div
        class="toggler-box__item "
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Test view of toggler Should show a toggler with "true" value 1`] = `
<DocumentFragment>
  <div
    class="toggler"
  >
    <div
      class="toggler-box"
    >
      <div
        class="toggler-box__item toggler-box__item_right"
      />
    </div>
  </div>
</DocumentFragment>
`;
